### PR TITLE
A2c adv fix

### PR DESCRIPTION
**Pull request by lgraesser**
1. Fixing n-step returns
2. Ensuring that the critic is trained to output V(s)

### CODE REVIEW

File: slm_lab/agent/algorithm/actor_critic.py [hunk 1]
```diff
@@ -314,15 +314,16 @@ def calc_gae_advs_v_targets(self, batch):
         Used for training with GAE
         '''
         v_preds = self.calc_v(batch['states'])
+        # v targets = r_t + gamma * V(s_(t+1))
+        next_v_preds = self.calc_v(batch['next_states'])
+        v_targets = math_util.calc_nstep_returns(batch, self.gamma, 1, next_v_preds)
         # calc next_state boundary value and concat with above for efficiency
-        next_v_pred_tail = self.calc_v(batch['next_states'][-1:])
+        next_v_pred_tail = next_v_preds[-1:]
         next_v_preds = torch.cat([v_preds[1:], next_v_pred_tail], dim=0)
```
Comment (kengz) [inline slm_lab/agent/algorithm/actor_critic.py]: now this is just next_v_preds. can be removed. line 319-321 can be removed

File: slm_lab/agent/algorithm/actor_critic.py [hunk 2]
```diff
@@ -314,15 +314,16 @@ def calc_gae_advs_v_targets(self, batch):
...
         # ensure val for next_state is 0 at done
         next_v_preds = next_v_preds * (1 - batch['dones'])
-
-        # v_targets = gae_targets + v_preds
         adv_targets = math_util.calc_gaes(batch['rewards'], v_preds, next_v_preds, self.gamma, self.lam)
-        v_targets = adv_targets + v_preds
+        # v_targets = adv_targets + v_preds
```
Comment (kengz) [inline slm_lab/agent/algorithm/actor_critic.py]: this can be removed too

File: slm_lab/agent/algorithm/actor_critic.py (hunk 1 above)
Comment (lgraesser) [reply slm_lab/agent/algorithm/actor_critic.py]: I see what this is doing now. :) Fixed.

File: slm_lab/agent/algorithm/actor_critic.py (hunk 2 above)
Comment (lgraesser) [reply slm_lab/agent/algorithm/actor_critic.py]: Done.

**Pull request merged**